subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -249,7 +250,10 @@ public DefaultProject(
 
         extensibleDynamicObject = new ExtensibleDynamicObject(this, Project.class, services.get(InstantiatorFactory.class).decorateLenient(services));
         if (parent != null) {
-            extensibleDynamicObject.setParent(parent.getInheritedScope());
+            DynamicObject parentInherited = services.get(CrossProjectModelAccess.class).parentProjectDynamicInheritedScope(this);
+            if (parentInherited != null) {
+                extensibleDynamicObject.setParent(parentInherited);
+            }
```

Comment:
We are now completely ignoring the parent reference, only using it to determine whether a project has a parent (and then determine the actual parent via CrossProjectModelAccess), right?

Maybe  add a comment or use a local `hasParent` or ìsSubproject` boolean value? Or just call CPMA blindly, as it makes the right choices regarding top-level projects.

💄

Reply:
Fixed, thanks!